sphinx_gallery/gen_rst.py

```diff
@@ -444,6 +454,37 @@ def save_thumbnail(image_path_template, src_file, gallery_conf):
         scale_image(default_thumb_file, thumb_file, 200, 140)
 
 
+def python_zip(file_list, target_dir, extension='.py'):
+    """Stores all files in file_list into an zip file
+
+    Parameters
+    ----------
+    file_list : list of str
+        Hold all the file names to be included in zip file
+    target_dir : str
+        name path of the directory where examples are being stored for
+        download in the gallery
+    extension : str
+        '.py' or '.ipynb' In order to deal with downloads of python
+        sources and jupyter notebooks the file extension from files in
+        file_list will be removed and replace with the value of this
+        variable while generating the zip file
+    Returns
+    -------
+    void : zip file is written as `target_dir_{python,jupyter}.zip`
+    depending on the extension
+"""
+    zipname = '_python' if extension == '.py' else '_jupyter'
+    zipname = target_dir + zipname + '.zip'
+
+    zipf = zipfile.ZipFile(zipname, mode='w')
```

Comment:
> I would use a with statement: "with ZipFile(...) as zipf:". The benefit is that if there is an exception, the file is closed. The drawback is that it only works in Python 2.7 and later.

I was aware of the context manager for this. But Sphinx-Gallery still
supports python 2.6 so I didn't use it. I remember I keep that support
since it was for your Neurodebian systems. Should I drop support for
Python 2.6? @lesteve suggested to drop support on Python 2.6 when NumPy
does it, which shall be anytime soon as it is announced for their 1.12
release.